client/cli/src/commands/runcmd.rs

```diff
@@ -359,13 +343,7 @@ impl RunCmd {
 		// set sentry mode (i.e. act as an authority but **never** actively participate)
 		config.sentry_mode = self.sentry;
 
-		config.offchain_worker = match (&self.offchain_worker, role) {
-			(OffchainWorkerEnabled::WhenValidating, sc_service::Roles::AUTHORITY) => true,
-			(OffchainWorkerEnabled::Always, _) => true,
-			(OffchainWorkerEnabled::Never, _) => false,
-			(OffchainWorkerEnabled::WhenValidating, _) => false,
-		};
-
+		self.offchain_worker.update_config(&mut config, role)?;
```

Comment:
You did it right but I got a new update on the cli that has just been push sorry :sweat_smile:  .... and it changes things a lot.

1. You will need to check the default implementation in client/cli/src/config.rs and adapt to your needs
2. In runcmd.rs you will need to change the runcmd implementation for that field like you did here (similarly)
3. You can keep the implementation in OffchainWorkerParams **BUT** you will need to rename update_config to offchain_worker and return the value (you don't receive config in parameter anymore)

Reply:
Thanks for the heads up!